project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -709,21 +764,62 @@ public Optional<String> retrieveRequestId() {
       return Optional.ofNullable(requestId);
     }
 
-    /**
-     * Returns the requestId, can be null.
-     *
-     * @return Returns the requestId, if no requestId is set, returns null.
-     */
-    @Override
-    public String requestId() {
-      return requestId;
-    }
-
     boolean hasRequestId() {
       return nonNull(requestId);
     }
   }
 
+  /**
+   * The actual request container for metadata validation.
+   * <p>
+   * A validation request contains a {@link ValidationRequestBody} with the actual metadata to be
+   * validated, next to the project ID and the request ID.
+   *
+   * @param projectId   the project ID of the project the metadata shall be validated for
+   * @param requestBody the actual metadata container with the information to be validated
+   * @param requestId   the request ID, needs to be provided by the client and will be referenced in
+   *                    the response. If <code>null</code> or {@link String#isBlank()} is true, then
+   *                    a random UUID is assigned with {@link UUID#randomUUID()}
+   * @since 1.10.0
+   */
+  record ValidationRequest(String projectId, ValidationRequestBody requestBody,
+                           String requestId) implements CacheableRequest {
+
+    public ValidationRequest {
+      if (projectId == null) {
+        throw new IllegalArgumentException("Project ID cannot be null");
+      }
+      if (projectId.isBlank()) {
+        throw new IllegalArgumentException("Project ID cannot be blank");
+      }
+      if (requestId != null && requestId.isBlank()) {
+        requestId = null;
+      }
+      if (requestId == null) {
+        requestId = UUID.randomUUID().toString();
+      }
```

Comment:
You can combine those if statements to avoid temporal coupling
```suggestion
      if (requestId == null || requestId.isBlank()) {
        requestId = UUID.randomUUID().toString();;
      }
```